testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/AuthenticationTestCase.java

```diff
@@ -103,13 +107,31 @@
                 .addClass(WhoAmI.class).addClass(Util.class).addClass(Entry.class)
                 .addClasses(WhoAmIServlet.class, AuthenticationTestCase.class)
                 .addClasses(AbstractSecurityDomainSetup.class, EjbSecurityDomainSetup.class)
+                .addClass(TestSuiteEnvironment.class)
                 .addAsResource(currentPackage, "users.properties", "users.properties")
                 .addAsResource(currentPackage, "roles.properties", "roles.properties")
                 .addAsWebInfResource(currentPackage, "web.xml", "web.xml")
                 .addAsWebInfResource(currentPackage, "jboss-web.xml", "jboss-web.xml")
                 .addAsWebInfResource(currentPackage, "jboss-ejb3.xml", "jboss-ejb3.xml")
                 .addAsManifestResource(new StringAsset("Manifest-Version: 1.0\nDependencies: org.jboss.as.controller-client,org.jboss.dmr\n"), "MANIFEST.MF")
-                .addAsManifestResource(currentPackage, AuthenticationTestCase.class.getSimpleName() + "-permissions.xml", "permissions.xml");
+                .addAsManifestResource(createPermissionsXmlAsset(
+                        // login module needs to modify pricipal to commit logging in
+                        new AuthPermission("modifyPrincipals"),
+                        // AuthenticationTestCase#testAuthenticatedCall calls org.jboss.security.client.JBossSecurityClient#performSimpleLogin
+                        new RuntimePermission("org.jboss.security.getSecurityContext"),
+                        new RuntimePermission("org.jboss.security.SecurityContextFactory.createSecurityContext"),
+                        new RuntimePermission("org.jboss.security.SecurityContextFactory.createUtil"),
+                        new RuntimePermission("org.jboss.security.plugins.JBossSecurityContext.setSubjectInfo"),
+                        new RuntimePermission("org.jboss.security.setSecurityContext"),
+                        // AuthenticationTestCase#execute calls ExecutorService#shutdownNow
+                        new RuntimePermission("modifyThread"),
+                        // AuthenticationTestCase#execute calls sun.net.www.http.HttpClient#openServer under the hood
+                        new SocketPermission(SERVER_HOST_PORT, "connect,resolve"),
+                        // TestSuiteEnvironment reads system properties
```

Comment:
These look ok. I think we should also consider adding the below properties to the security manager subsystem as part of the test suit setup. As they are clearly test suite specific their presence there will not mask real bugs.